### ISSUE TITLE
Subscription without post

### ISSUE THREAD

**Issue by ThierryNK** [stcr/subscribe-to-comments-reloaded#645]
Hi
Link that is emailed for a subscription without post goes to the Manage subscription page that is asking g again for an email.
Odd

You can test here: https://audiophile-magazine.com/discographie/1er-concerto-pour-piano-tchaikovsky-alexander-malofeev/

**Comment by BobaWebDev** [comment]
Hi @ThierryNK 

I answered on the WordPress.org support request, but let's continue here, you can disregard the one on WordPress.org forum.

Can you please try disabling the challenge question and then give it a try. Just temporary disable it for a test. 

Let me know if it works out that way, if it does then I'll know where in the code to look for the issue.

**Comment by BobaWebDev** [comment]
Hi @ThierryNK 

Sorry for the wait, weekend got in the way.

Let's first test if the page knows that it's supposed to be confirming a subscription. 

On line 744 of wp_subscribe_reloaded.php you'll see:

`$email_by_key = $this->utils->get_subscriber_email_by_key($srek);`

After that line just add:

`var_dump( $action, $key );`

Then try to confirm a subscription and let me know what are the 2 values that are shown at the very top of the page.

**Comment by ThierryNK** [comment]
string(1) "s" string(32) "9219b947e970f04c3658f434e250c8d2"

**Comment by ThierryNK** [comment]
And when confirming the subscription:

int(0) int(0)

**Comment by BobaWebDev** [comment]
Yeah, it doesn't know it's on a confirmation page, which is weird. 

I tried making another comment to check the URL I get for confirmation, but I'm getting an error saying captcha response is incorrect.

It says `onze + cinq`, the answer should be `16`, right?

**Comment by ThierryNK** [comment]
Hi
There was something going bad with the safety challenge question. 

Now the mail that I receive is not an url, it starts with
x-webdoc://F856DA9D-D279-4BC8-BDFE-57016816BEFF/'https://audiophile-magazine.com/comment-subscriptions/

Other point, "Check your email to confirm your subscription." that appears when posting a comment, cannot be translated.

So, sorry, I will disable your plugin. I cannot make tests, leave debug messages, send unusable  mails and not fully translated strings on a living website.

You can post here if you have any news.

Regards

**Comment by BobaWebDev** [comment]
Nothing specific comes to mind as the cause, so testing is the only thing that will lead to some answers, sorry.

If you can test, the next test would be to install [WP Mail Logging](https://wordpress.org/plugins/wp-mail-logging/) plugin and then try a subscription. After that go to the mail log (WP admin > Tools > WP Mail Log) and try the confirmation URL from there.     That would help point the debugging in the right direction.

**Comment by ThierryNK** [comment]
When I have more time, I will setup a sandbox website
Meanwhile, if you want to do some tests, the configuration is the following, server Centos 94.0.7, Apache 2.4.46, ,php 7.4, Theme GeneratePress, Wordpress 5.7.1,
